### ISSUE TITLE
Simpler way for getting all fields of a model populated?

### ISSUE THREAD

**Issue by vedtam** [redux-orm/redux-orm#335]
Hi,

Is there a simple way of flattening a model without the need to access every related property to get their value?

What I would like to simplify:

```
export const channelStories = (channelId: number) => <any>ormCreateSelector(
  orm,
  (session) => {
    const ch = session.Channel.withId(channelId);
    if (ch) {
      return {
        ...ch.ref, stories: ch.stories.all().toRefArray(),
        ...ch.ref, member: ch.member.ref
       // 10+ more relations
      };
    }
  }
);
```

thanks!

**Comment by kirbydcool** [comment]
I don't believe there's currently a supported way of doing this unfortunately. It may be possible if you dive into the redux-orm internals since the models do keep track of relationships, but that's not publicly supported.

As an alternative, would it make sense to not include the related models and query them with separate selectors when you need them? I can't imagine every use of channel needs access to all of the relations.

**Comment by haveyaseen** [comment]
Exactly my thoughts too, especially because if you include all relationships in the selector then the selector will always need to be re-run as soon as one of the relationships changes in some way. Which you often don't care about.

Apart from that, it's usually not a good idea to create selectors within other functions.

**Comment by samuelcolburn** [comment]
I asked a similar question a few months ago ([thread](https://github.com/redux-orm/redux-orm/issues/269#issuecomment-508285450)), so that may give you some insights. I ended up reworking a lot of my components to use more specific selectors rather than flattening everything. The only exception I currently have is one where I need to join them to apply a sort that requires values from both entities. I don't know if it's the best way, but this is how I did it, after getting some insights from @haveyaseen in the aforementioned issue:

```js
// I just get one task at a time here, combined with its area parent values
export const taskWithArea = ormCreateSelector(
  orm.task,
  orm.task.area,
  (task, area) => ({ ...task, area })
)

// Then I'll compose that with another selector if I need to
export const tasksWithArea = ormCreateSelector(
  orm.User.tasks.map(taskWithArea),
  customSortFn
)
```
